torch/_inductor/optimize_indexing.py

```diff
@@ -136,77 +156,105 @@ def constant(value, dtype):
 
     @staticmethod
     def reciprocal(x):
-        return ValueRanges.checked_unary_map(x, lambda y: 1 / y)
+        x = ValueRanges.wrap(x)
+        if 0 in x:
+            return ValueRanges(-math.inf, math.inf)
+        else:
+            return ValueRanges.decreasing_map(x, lambda y: 1 / y)
```

Comment:
would be simpler as `V.get_ops_handler().div(1, y)` i think

Reply:
What do you mean exactly?

Reply:
reciprocal is `div(1, x)`, we could reuse div instead of duplicating the function

Reply:
ah, the thing is that this specialisation gives simpler expressions for the case of symbolic shapes, that's why I kept it